### PR TITLE
[ui] cleanup natives.dart

### PR DESCRIPTION
**Pull request by jonahwilliams**
I added types and breadcrumbs to some methods here. Also deleted some historic code that is no longer required (deleted in https://dart-review.googlesource.com/c/sdk/+/129742 )

### CODE REVIEW

File: lib/ui/natives.dart
Comment (jonahwilliams) [inline lib/ui/natives.dart:82]: I can find no references to this in the engine or in the dart/sdk. Since it does not have a pragma it is almost certainly getting tree shaken

Comment (jonahwilliams) [reply lib/ui/natives.dart:82]: https://github.com/flutter/engine/pull/2762

**Comment by jonahwilliams** [comment]
Waiting on some confirmation that this code is safe to delete

File: lib/ui/natives.dart
Comment (jonahwilliams) [reply lib/ui/natives.dart:82]: Confirmed this was removed from the Dart SDK in https://dart-review.googlesource.com/c/sdk/+/129742 and is safe to delete

**Review by dnfield — APPROVED** [review]
LGTM once it has a test exemption from @Hixie

**Comment by Hixie** [comment]
test-exempt: code refactor with no semantic change

**Pull request merged**
